### PR TITLE
feat: bundle expect from @playwright/test library

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ fix [#808 ](https://github.com/elastic/synthetics/issues/700)
+ Bundles the PW test library expect matcher with the synthetics and replaces the expect from jest in favor of the PW test matchers. 
+ The PR uses the `expect` library from the `@playwright/test` without bringing in the full runner which we do not require as we have our own Synthetics runner which is build with different syntax. This also replaces the `expect` from jest in favor of the PW test matchers which under the hood is built on top of expect.
+ With bundling approach, we get the benefit of only importing the necessary modules from the PW test and core and still utilize all the available matchers except for the matchers that does comparision on the files - `toMatchSnapshot` and `toHaveScreenshot`. 

### Folowup
+ [ ] Update the docs around expect library - https://www.elastic.co/guide/en/observability/8.9/synthetics-create-test.html#synthetics-assertions-methods
+ [x] Callout `toHaveScreenshot` and `toMatchSnapshot` wont work. 
+ [ ] Stacktrace fix to polish the errors from Expect - #824

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,9 +6,10 @@
   "types": "dist/index.d.ts",
   "scripts": {
     "clean": "rimraf dist",
-    "prepublish": "npm run clean && npm run build",
+    "prepublish": "npm run clean && npm run build:bundles && npm run build",
```
Comment (shahzad31) [inline package.json:9]: seems like it will run npm run build twice in same directory?

Comment (vigneshshanmugam) [reply package.json]: Fixed.

**Review by devcorpio — APPROVED** [review]
this looks great! besides the code is clean and easy to follow! 🚀 

just a question related to the PR's description:

You mentioned there that "toHaveScreenshot and toMatchSnapshot" will not work. But when testing your PR, I was able to see them within the playwright expect module:

playwright-expect

I'm sure that what you mentioned is for a particular reason. Just out of curiosity, why are they not going to work?

**Comment by shahzad31** [comment]
> LGTM! This looks great! besides, the code is clean and easy to follow! 🚀
> 
> just a question related to the PR's description:
> 
> You mentioned that "toHaveScreenshot and toMatchSnapshot" will not work. But when testing your PR, I was able to see them within the playwright expect module:
> 
> playwright-expect
> I'm sure that what you mentioned is for a particular reason. Just out of curiosity, why are they not going to work?

we need some typescript magic to remove those assertions or some other magic to throw an error when user tries to use those assertions.

**Review by shahzad31 — APPROVED** [review]
LGTM, 

Looks great, i pushed a small change for non supported assertion !!

**Comment by shahzad31** [comment]
image

**Comment by devcorpio** [comment]
> Looks great, i pushed a small change for non supported assertion !!

Nice idea! 🎉 

> Somehow for me types aren't properly highlighted for used assertions


hmm, in vscode seems to work properly. Is perhaps a webstorm "problem"? Do you need to restart or something?

**Comment by vigneshshanmugam** [comment]
> Somehow for me types aren't properly highlighted for used assertions

You need to nuke some node_modules files and do build again for it to work correctly with new types. 

> Looks great, i pushed a small change for non supported assertion !!

Thanks, I forgot to push it. I am working on a small TS change which might show it might not work. I will take your change if its not working.

**Comment by devcorpio** [comment]
> I am not sure how its working for you. 

I meant that the IDE IntelliSense with typescript was working correctly for me(Shahzad could write things like expect(page).toHaveInventedAssertion())

**Comment by vigneshshanmugam** [comment]
I see, typings would work, but the method itself wont work which Is why we have to exclude them or show errors.

**Comment by paulb-elastic** [comment]
https://github.com/elastic/observability-docs/issues/3206 raised for the docs update

**Pull request merged**